crates/oxc_linter/src/rules/unicorn/switch_case_braces.rs

```diff
@@ -142,8 +146,11 @@ impl Rule for SwitchCaseBraces {
             };
 
             if self.always_braces && missing_braces {
-                let colon = u32::try_from(ctx.source_range(case.span).find(':').unwrap()).unwrap();
-                let span = Span::sized(case.span.start, colon + 1);
+                let colon_end = u32::try_from(
+                    switch_clause_regex.find(ctx.source_range(case.span)).unwrap().end(),
```

Comment:
rather than using regex, can we try searching from the end of the test? e.g. `case.test.span().end`?

that should put us 
```
case "foo:bar".  :
              ^
```

then we can start searching forward

Reply:
I understood the idea, however not sure about how to use `case.test.span().end` as `case.test` is an expression.

So, I found another way to extract only the wanted part:
```
case "scope:type": doSomething();
                   ^ case.consequent[0].span().start
^ case.span().start
```
Then I do a `rfind` on the colon.

I just do not know if accessing the `case.consequent[0]` is a safe choice or not.